### PR TITLE
[TTNNFusing] Fix attention mask broadcast for SDPA

### PR DESCRIPTION
**Pull request by mvasiljevicTT**
### Ticket
[xla pcc issue](https://github.com/tenstorrent/tt-xla/issues/4206)
[known metal sdpa issue](https://github.com/tenstorrent/tt-metal/issues/39910)

### Problem description
Even though pcc on prefill is good, outputs for multichip LLMs are mostly bad, like in [this nightly run](https://github.com/tenstorrent/tt-xla/actions/runs/24642681050/job/72084379748). This is especially present in galaxy llama and gpt oss.

### What's changed
The root cause is attention_mask broadcasting in the decode sdpa op. 

Without checking and asserting shape, device ops assumes that it has same batch and num_heads as Q. When provided with tensor of batch_size 1 instead, it still reads rest of the data in expected size. When these are all 0, and KV cache has 0 on future places,  it might not look that bad with outputs repeating sensible frequent words, but if something else is already written there output can be completely random. 

Output for the first user was always ok as it was the only one with correct attention_mask. The fix was needed only for decode, so pcc that we calculate on prefill will not be affected.

Some perf drop is expected, but it will be back once we support `is_causal` and `sliding_window` built-in masks.

It should not change pcc of any model as we measure it today because we check only first user prefill, and this is fixing decode for non-first tokens. 

### Checklist
- [x] [perf benchmark on llmbox](https://github.com/tenstorrent/tt-xla/actions/runs/24664607317/job/72120802836) - mlir change only (gpt oss still expected to be weird)
- [x] [perf benchmark on galaxy](https://github.com/tenstorrent/tt-xla/actions/runs/24666905318/job/72129144995) - with [xla fix for gpt oss](https://github.com/tenstorrent/tt-xla/pull/4316)
- [x] [perf benchmark n150](https://github.com/tenstorrent/tt-xla/actions/runs/24710067795)

### CODE REVIEW

**Comment by mvasiljevicTT** [comment]
cc @odjuricicTT

**Review by odjuricicTT — APPROVED** [review]
Nice!

This will affect single chip models as well? Can you run benchmarks for those?

**Review by mtopalovicTT — APPROVED** [review]

**Comment by mvasiljevicTT** [comment]
> Nice!
> 
> This will affect single chip models as well? Can you run benchmarks for those?

Sure, added now

**Review by ndrakulicTT — APPROVED** [review]
`tools/golden/mapping.py` looks good
